packages/get-platform/src/getPlatform.ts

```diff
@@ -132,20 +133,26 @@ export function parseOpenSSLVersion(input: string): string | undefined {
 }
 
 // getOpenSSLVersion returns the OpenSSL version excluding the patch version, e.g. "1.1.x"
-export async function getOpenSSLVersion(): Promise<string | undefined> {
+export async function getOpenSSLVersion(platform: NodeJS.Platform, arch: Arch): Promise<string | undefined> {
+  let prefix: string | undefined = undefined
+
+  if (platform == 'android') {
+    prefix = '/data/data/com.termux/files/usr'
+  }
+
+  const isArmAndAndroid = arch === 'arm' || arch === 'arm64' && platform == 'android'
   const [version, ls] = await Promise.all([
     gracefulExec(`openssl version -v`),
     gracefulExec(`
-      ls -l /lib64 | grep ssl;
-      ls -l /usr/lib64 | grep ssl;
+      ls -l ${prefix ? prefix : ''}/lib${isArmAndAndroid ? '' : '64'} | grep ssl;
+      ls -l ${prefix ? prefix : '/usr'}/lib${isArmAndAndroid ? '' : '64'} | grep ssl;
```

Comment:
1. The second line is still unnecessary in the case of Android.

2. Re: `isArmAndAndroid` check — this is actually a great point that the library directories are different on different platforms, and even not counting the Android what we have here is not always correct for all the platforms that we support.

The best approach here would be to have an array of the library paths that depends on the platform and architecture (and optionally distro, but mixing `debian`-specific and `redhat`-specific paths together is also fine), and then doing something like ``libraryPaths.map(path => `ls -l ${path} | grep ssl;`).join('\n')`` here.

So you'd have all of these checks separately, and an array like `['/data/data/com.termux/files/usr/lib']` for Android/Termux, `['/lib64', '/usr/lib64', '/lib/x86_64-linux-gnu/', '/usr/lib/x86_64-linux-gnu/', '/lib', '/usr/lib']` for Linux on x86, `['/lib64', '/usr/lib64', '/lib/aarch64-linux-gnu/', '/usr/lib/aarch64-linux-gnu/', '/lib', '/usr/lib']` for Linux on ARM64, etc.

Reply:
I dont think termux have lib64 folder in x64

Reply:
@CanKolay3499 sure, and I didn't say it had. On the contrary, what I'm suggesting is, among other things, to only check one single path in Termux (`/data/data/com.termux/files/usr/lib`)

Reply:
> The second line is still unnecessary in the case of Android.

Fixed

> The best approach here would be to have an array of the library paths that depends on the platform and architecture (and optionally distro, but mixing debian-specific and redhat-specific paths together is also fine), and then doing something like libraryPaths.map(path => `ls -l ${path} | grep ssl;`).join('\n') here.
So you'd have all of these checks separately, and an array like ['/data/data/com.termux/files/usr/lib'] for Android/Termux, ['/lib64', '/usr/lib64', '/lib/x86_64-linux-gnu', '/usr/lib/x86_64-linux-gnu', '/lib', '/usr/lib'] for Linux on x86, ['/lib64', '/usr/lib64', '/lib/aarch64-linux-gnu', '/usr/lib/aarch64-linux-gnu', '/lib', '/usr/lib'] for Linux on ARM64, etc.

Out of this pulls scope.